src/lsp/MessageHandlers.re

```diff
@@ -148,32 +145,37 @@ let handlers: list((string, (state, Json.t) => result((state, Json.t), string)))
     open InfixResult;
     let%try (uri, pos) = Protocol.rPositionParams(params);
     let%try package = State.getPackage(uri, state);
-    let%try_wrap (file, extra) = State.fileForUri(state, ~package, uri) |> Result.orError("Could not compile " ++ uri);
+    let%try_wrap (file, extra) = State.fileForUri(state, ~package, uri) |> orError("Could not compile " ++ uri);
 
     open Infix;
 
     {
       let%opt (_, loc) = References.locForPos(~extra, Utils.cmtLocFromVscode(pos));
-      let allModules = package.localModules |> List.map(fst);
+      let allModules = List.map(package.localModules, fst);
       let%opt allReferences = References.forLoc(
         ~file,
         ~extra,
         ~allModules,
         ~getModule=State.fileForModule(state, ~package),
         ~getExtra=State.extraForModule(state, ~package),
         loc
-      ) |> Result.toOptionAndLog;
+      ) |> toOptionAndLog;
 
       open Rpc.J;
       Some((
         state,
-        l(allReferences |> List.map(
-          ((fname, references)) => (
-            fname == uri
-            ? List.filter(((loc)) => !Protocol.locationContains(loc, pos), references)
-            : references
-          ) |> List.map(((loc)) => Protocol.locationOfLoc(~fname, loc))
-        ) |> List.concat)
+        List.map(
+          allReferences,
+          ((fname, references)) => {
+            let locs = fname == uri
+              ? List.keep(references, loc => !Protocol.locationContains(loc, pos))
+              : references;
+            List.map(locs, loc => Protocol.locationOfLoc(~fname, loc))
+          }
+        )
+        |. List.toArray
+        |. List.concatMany
```

Comment:
umm why does belt's list have a function that works on arrays?

Reply:
@jaredly Belt's conversion function stays on the original data structure. Like converting from list to array => List.toArray .

Reply:
Yeah. Can’t have bidirectional conversion because that’s a dependency loop. Pulling it out into a Convert module would work, maybe.

Reply:
It's also weird to me that you have to convert to an array before `concat`ing :/ why is `List` concating arrays anyway? but this is fine I guess

Reply:
It's an idiom for concatMany in all modules to use array; map and set do this too. Put in other words, map's concatMany wouldn't take a map as the variadic argument.

We can reconsider this though?

Reply:
hmmm interesting. it sure is weird to have to convert a list of lists to an array of lists just to concat them, maybe there can be an extra method there